src/core/lib/slice/slice_utils.h

```diff
@@ -101,15 +107,16 @@ struct ManagedMemorySlice : public grpc_slice {
   explicit ManagedMemorySlice(const char* string);
   ManagedMemorySlice(const char* buf, size_t len);
   explicit ManagedMemorySlice(const grpc_slice* slice);
-  bool Equals(const grpc_slice& other) const {
+  bool operator==(const grpc_slice& other) const {
     if (refcount == other.refcount) {
       return true;
     }
     return !grpc_slice_differs_refcounted(other, *this);
   }
-  bool Equals(const char* buf, const size_t len) const {
-    return data.refcounted.length == len && buf != nullptr &&
-           memcmp(buf, data.refcounted.bytes, len) == 0;
+  bool operator!=(const grpc_slice& other) const { return !(*this == other); }
+  bool operator==(std::pair<const char*, size_t> buflen) const {
```

Comment:
I don't think it's valid to have `operator==()` with more than one argument, because the syntax for the `==` operator can only accept one additional operand (other than `this`).  If we need this version, then we may have to go back to just having an `Equals()` method, at least for this variant. :(

Reply:
True that it only allows one argument. However, this pair<> counts as one argument.

Reply:
Done.

Reply:
Ironicly, Esun just yesterday finished the changes that now allow us to use the C++ standard library from within core, and just this morning he merged #20440, which replaces `grpc_core::Pair<>` with direct uses of `std::pair<>`.  So, sorry, but you'll have to back out that particular change. :(

Reply:
Ah, nevermind, I see you just did that. :)